### PR TITLE
Put built dlls in lib/scipp and copy from that location

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
WIP; test on CI

### CODE REVIEW

File: cmake/scipp-install.cmake [hunk 1]
```diff
@@ -10,7 +10,11 @@ function(scipp_install_component)
   )
   add_sanitizers(${SCIPP_INSTALL_COMPONENT_TARGET})
   if(DYNAMIC_LIB)
-    install(TARGETS ${SCIPP_INSTALL_COMPONENT_TARGET} EXPORT ${EXPORT_NAME})
+    install(
+      TARGETS ${SCIPP_INSTALL_COMPONENT_TARGET}
+      EXPORT ${EXPORT_NAME}
+      RUNTIME DESTINATION Lib/scipp
```
Comment (SimonHeybrock) [inline cmake/scipp-install.cmake:16]: Is this flag used only by Windows? Otherwise it would break OSX and Linux?

Comment (Tom-Willemsen) [reply cmake/scipp-install.cmake:16]: According to https://cmake.org/cmake/help/latest/command/install.html it affects executables (we don't install any as far as I can see) and DLLs (only on windows systems). On linux and osx I believe cmake uses `LIBRARY DESTINATION ...`

File: tools/build_conda.py [hunk 2]
```diff
@@ -50,7 +50,7 @@ def move(self, src, dst):
     # Depending on the platform, directories have different names.
     if sys.platform == "win32":
         lib_dest = 'lib'
-        dll_src = 'bin'
+        dll_src = os.path.join("lib", "scipp")
         dll_dest = os.path.join("bin")
```
Comment (SimonHeybrock) [inline tools/build_conda.py:54]: Need to revert this line that I changed this morning?

Comment (Tom-Willemsen) [reply tools/build_conda.py:54]: I think this is correct as the DLLs are now being built in a different location, so we need to copy from that new location?

Comment (SimonHeybrock) [reply tools/build_conda.py:54]: Yes `dll_src` is correct, but `dll_dest` needs to revert (you may not have this commit locally) to not use `"bin"`.

**Pull request merged**